src/index.test.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  BatchGetCommand,
+  BatchGetCommandInput,
+  BatchGetCommandOutput,
+  BatchWriteCommand,
+  BatchWriteCommandInput,
+  BatchWriteCommandOutput,
+  DeleteCommand,
+  DeleteCommandOutput,
+  DynamoDBDocumentClient,
+  GetCommand,
+  GetCommandOutput,
+  PutCommand,
+  PutCommandOutput,
+} from '@aws-sdk/lib-dynamodb';
+import { mockClient } from 'aws-sdk-client-mock';
+import { MhDynamoDbClient } from '.';
+
+describe('test client', () => {
+  describe('constructor', () => {
+    it('should call awsClientCapture with a new instance of DynamoDb if options.awsClientCapture is provided', () => {
+      const mockedFunction = vi.fn((x) => x);
+
+      const instance = new MhDynamoDbClient(
+        { awsClientCapture: mockedFunction },
+        {}
+      );
+
+      expect(mockedFunction).toHaveBeenCalled();
+    });
+
+    it('should not call awsClientCapture if options.awsClientCapture is not provided', () => {
+      const mockedFunction = vi.fn((x) => x);
+
+      const instance = new MhDynamoDbClient({});
+
+      expect(mockedFunction).not.toHaveBeenCalled();
+    });
+  });
+
+  it('should call getItem and return the result', async () => {
+    const mockResponse: GetCommandOutput = {
+      Item: { id: '123', name: 'Test Item' },
+      $metadata: {},
+    };
+
+    const mhDynamoDbClient = new MhDynamoDbClient();
+    const mock = mockClient(DynamoDBDocumentClient);
```

Comment:
looks weird imo that we mock after we initialize our class? I would expect it to be mocked before so that our class will be initialized with a mocked `client`... 🤷‍♂️